### ISSUE TITLE
Help including virtual hook

### ISSUE THREAD

**Issue by decapon** [binary-com/binary-bot#3745]
Hi

Could anyone help me introduce virtual hook into a particular binary bot I use? 

Will really appreciate the assistance

**Comment by binary-ex-machina** [comment]
You can use the virtual hook without any additional code. All you need is to load your script at https://binaryblocks.pro/ and modify this section in the first block, changing these parameters and inserting a token from an account that will make the virtual trades

![image](https://user-images.githubusercontent.com/34610787/196065340-12c9ff8d-0ccf-4a81-b71c-492fa02b6bee.png)

**Comment by decapon** [comment]
Hi.

Thanks for response and sorry for the delay in my acknowledgement. Was indisposed and couldn't log in till now.

I'm not exactly clear on what you mean by inserting a token from an account. I've tried inputting various values in that parameter and all lead to the bot failing to run.

Could you kindly enlighten further?

Million thanks

**Comment by binary-ex-machina** [comment]
The token you need to insert must be from an account other than yours. You can create an account with another e-mail to use the demo. After you have the account, generate the account token and use it in the token input field

**Comment by decapon** [comment]
Hi 

Okay I've been able to get the token from another account.

But the virtual hook isn't performing as it should. Once it switches back to a normal trade after a loss in virtual, it uses a stake on the next martingale level, just as if all the virtual trades were normal trades.

The point of wanting the virtual hook was to protect the normal account from a losing streak and preserve the martingale steps. And this is defeated with the way this operates

Refer to the screen shot below. You will see that after the consecutive virtual losses of 0.35, when it entered the normal trade, it took a trade of 1.54! which is the third martingale level. It should have started at the first level of 0.35 since it was the very first trade taken on the real account

![Screenshot_20221024-093238](https://user-images.githubusercontent.com/115776743/197484332-8ccdfd20-8369-4962-9f7c-8a08482d1bab.jpg)

**Comment by binary-ex-machina** [comment]
@decapon, that happened because you applied martingale on any trade. You must apply martingale only in the trades that aren't virtual. You must use a block to check if the actual trade is a virtual trade. If it's not, you apply martingale. This way, the martingale will change only when the trade is not a virtual trade. Check the example below.

![image](https://user-images.githubusercontent.com/34610787/197843877-88e6371c-9291-46f7-aa39-4b6e52ffe2d5.png)

**Comment by decapon** [comment]
Okay. What block do I use to effect those settings? I searched under money
management and saw a sub block named martingale but upon clicking on it, it
didn't expand as depicted in your shot.

Many many thanks again for your assistance.

**Comment by binary-ex-machina** [comment]
@decapon, try using a desktop device and see if it appears like this

![image](https://user-images.githubusercontent.com/34610787/198676706-d129ab0b-c072-4c1c-a2ac-62a678a09efa.png)

**Comment by decapon** [comment]
Yeah that's exactly how it appears. But it's different from the earlier
shot you sent and there are no sections where I can designate martingale
application for real trades only as shown in your shot
